src/elements/element.line.js

```diff
@@ -145,35 +363,127 @@ class Line extends Element {
 		super(props);
 	}
 
-	draw(ctx) {
+	set points(points) {
+		this._points = points;
+		delete this._segments;
+	}
+
+	get points() {
+		return this._points;
+	}
+
+	get segments() {
+		return this._segments || (this._segments = computeSegments(this));
```

Comment:
I worry that creating this new `_segments` array would be quite expensive because we're creating a new object for every data point. Can we continue to use the existing points as before instead of creating a new data structure? For the simple case (no bezier, etc.) we already have everything we need and should be able to avoid computing anything new

Reply:
Good worry, but the normal case (withou NaN data points). Will have only one segment. Also the drawing part does not have to check that now so this should be faster in subsequent draws. Not by noticeable amount through.